### PR TITLE
Check size of scratchpad before allocating memory and creating scratchpad.

### PR DESCRIPTION
**Pull request by ayzhuang**

### CODE REVIEW

File: src/ngraph/runtime/cpu/builder/add.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace ngraph
 
                     auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
                     auto sum_pd = mkldnn_emitter->get_elementwise_add_desc(node);
-                    QUERY_SCRATCHPAD(sum, sum_pd);
+                    size_t s_size = QUERY_SCRATCHPAD(sum, sum_pd);
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/builder/add.cpp]: s_* is used for static variables. Rename to something like 'scratchpad_size' or 'tmpbuf_size'

File: src/ngraph/runtime/cpu/builder/quantization.cpp [hunk 2]
```diff
@@ -62,7 +63,7 @@ namespace ngraph
                     {
                         auto arg1_buffer_index =
                             external_function->get_buffer_index(args[1].get_name());
-                        auto scales_size = shape_size(args[1].get_shape());
+                        auto scalescratchpad_size = shape_size(args[1].get_shape());
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/builder/quantization.cpp]: scales_size

File: src/ngraph/runtime/cpu/builder/quantization.cpp [hunk 3]
```diff
@@ -325,15 +336,16 @@ namespace ngraph
...
+                        QUERY_SCRATCHPAD_2ARGS(reorder, input_desc, result_desc);
 
                     auto scale_const_op =
                         std::dynamic_pointer_cast<ngraph::op::Constant>(quantize->get_argument(1));
                     if (scale_const_op == nullptr)
                     {
                         auto arg1_buffer_index =
                             external_function->get_buffer_index(args[1].get_name());
-                        auto scales_size = shape_size(args[1].get_shape());
+                        auto scalescratchpad_size = shape_size(args[1].get_shape());
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/builder/quantization.cpp]: scales_size

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 4]
```diff
@@ -685,9 +685,16 @@ using namespace ngraph::runtime;
     writer << "mkldnn_scratchpad_mds = std::vector<mkldnn::memory::desc*>("
            << to_string(m_mkldnn_emitter->get_mkldnn_scratchpad_mds().size()) << ");\n";
     writer << "size_t scratchpad_size = " << m_mkldnn_emitter->get_max_scratchpad_size() << ";\n";
+    writer << "if (scratchpad_size > 0)\n";
+    writer.block_begin();
```
Comment (gaurides) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Do we need these nested writer.block_begin()  / end() ?

Comment (ayzhuang) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: yes, it writes "{}" and  adds indent.

**Pull request merged**